### ISSUE TITLE
Icon Request: ElectronMail + Fix tray icon

### ISSUE THREAD

**Issue by Dvlnst** [numixproject/numix-core#5044]
| Application name | Icon name | Desktop file name | Original icon |
| --- | :-- | :-- | :-- |
| `ElectronMail`| `electron-mail` | `electron-mail` | ![original](https://raw.githubusercontent.com/vladimiry/ElectronMail/master/src/assets/dist/icons/png/64x64.png) |

Description: `ElectronMail is an Electron-based unofficial desktop client for ProtonMail. The app aims to provide enhanced desktop user experience enabling features that are not supported by the official in-browser web clients.

## Features

- :octocat: **Open Source**.
- :gear: **Reproducible builds**.
- :gear: **Cross platform**. The app works on Linux/OSX/Windows platforms.
- :mag_right: **Full-text search**. Enabled with [v2.2.0](https://github.com/vladimiry/ElectronMail/releases/tag/v2.2.0) release. See the respective [issue](https://github.com/vladimiry/ElectronMail/issues/92) for details.
- :package: **Offline access to the emails**. The [local store](https://user-images.githubusercontent.com/1560781/51189497-382a6c00-18f1-11e9-9b9a-baa63f0c0ff4.gif) feature enables storing your messages in the encrypted `database.bin` file, so you could view your messages offline, perform a full-text search against them and export them to EML files. Enabled since [v2.0.0](https://github.com/vladimiry/ElectronMail/releases/tag/v2.0.0) release.
- :mailbox: **Multi accounts** support per each email provider including supporting individual [entry point domains](https://github.com/vladimiry/ElectronMail/issues/29).
- :unlock: **Automatic login into the app** with a remembered master password using [keytar](https://github.com/atom/node-keytar) module ([keep me signed in](images/keep-me-signed-in.png) feature).
- :unlock: **Automatic login into the email accounts**, including filling [2FA tokens](https://github.com/vladimiry/ElectronMail/issues/10). Two auto-login delay scenarios supported in order to make it harder to correlate the identities, see the respective [issue](https://github.com/vladimiry/ElectronMail/issues/121).
- :unlock: **Persistent email account sessions**. The feature introduced since [v4.2.0](https://github.com/vladimiry/ElectronMail/releases/tag/v4.2.0) version with the `experimental` label, [#227](https://github.com/vladimiry/ElectronMail/issues/227). The feature enables the scenario when you to enter the account credentials on the login form only once, manually or automatically by the app, and then you never see the login form anymore for this email account even if you restart the app (unless you explicitly dropped the session in the admin area or it got dropped by the service due to the inactivity/expiration). If this feature is enabled for the account, manual credentials filling is the preferred option as a more secure option since you don't save the account credentials anywhere (`credentials` are encrypted though event if saved, see `settings.bin` file description in the [FAQ](https://github.com/vladimiry/ElectronMail/wiki/FAQ)).
- :closed_lock_with_key: **Encrypted local storage** with switchable predefined key derivation and encryption presets. Argon2 is used as the default key derivation function.
- :bell: **Native notifications** for individual accounts clicking on which focuses the app window and selects respective account in the accounts list.
- :bell: **System tray icon** with a total number of unread messages shown on top of it. Enabling [local messages store](https://github.com/vladimiry/ElectronMail/issues/32) improves this feature ([how to enable](https://github.com/vladimiry/ElectronMail/releases/tag/v2.0.0-beta.1)), see respective [issue](https://github.com/vladimiry/ElectronMail/issues/30).
- :gear: **Starting minimized to tray**.
- :gear: **Closing to tray**.
- :gear: **Switchable accounts handle buttons positioning** (vertical, horizontal and horizontal-dropdown). See details [here](https://github.com/vladimiry/ElectronMail/issues/36) and screenshots in the [images](images) folder.
- :package: **Batch emails export** to EML files. Feature released with [v2.0.0-beta.4](https://github.com/vladimiry/ElectronMail/releases/tag/v2.0.0-beta.4) version, requires `local messages store` feature to be enabled ([how to enable](https://github.com/vladimiry/ElectronMail/releases/tag/v2.0.0-beta.1)).
- :closed_lock_with_key: **Built-in/prepackaged web clients**. The built-in web clients are built from source code, see respective official [Protonmail](https://github.com/ProtonMail/WebClient) repository. See [79](https://github.com/vladimiry/ElectronMail/issues/79) and [80](https://github.com/vladimiry/ElectronMail/issues/80) issues for details.
- :gear: **Configuring proxy per account** support. Enabled since [v3.0.0](https://github.com/vladimiry/ElectronMail/releases/tag/v3.0.0) release. See [113](https://github.com/vladimiry/ElectronMail/issues/113) and [120](https://github.com/vladimiry/ElectronMail/issues/120) issues for details.
- :pencil: **Spell Checking**.
`

Website: `https://github.com/vladimiry/ElectronMail`

By default, the tray icon is being override by electron icon. Could you fix it?

Kind regards.

**Comment by palob** [comment]
Could you please research the tray icon name or post a screenshot?

**Comment by Dvlnst** [comment]
I don't know exactly what you mean with research the tray icon name. If there is a way to check it, tell me, I'm happy to help. A workaround I'm using right now:

The problem is related to default electron icon (electron.svg):
![Screenshot from 2020-02-04 00-37-45](https://user-images.githubusercontent.com/36289254/73700858-fa5cd900-46e7-11ea-918b-27afb0eb4f80.png)

I don't need it right now, so removing it does the trick:
`$ sudo mv /usr/share/icons/Numix-Circle/48/apps/electron.svg /usr/share/icons/Numix-Circle/48/apps/electron.svg_backup`
`$ sudo update-icon-caches /usr/share/icons/*`

![Screenshot from 2020-02-04 00-39-19](https://user-images.githubusercontent.com/36289254/73700897-25472d00-46e8-11ea-8f2b-d9d854722fbd.png)

**Comment by palob** [comment]
Well, if it just uses ``electron`` as icon name we probably can't fix it without further ado. You'll probably need a fixing script, I don't know if there's a way to have it pick up a tray icon with ``electronmail-tray`` for instance.
But then again I haven't got the slightest notion of how tray icon theming is done. Personally I try to get rid of tray icons as much as possible.

**Comment by creepertron95** [comment]
The Papirus icon theme has a tray icon for this program. The tray icon for ElectronMail is hardcoded.

The tray icon was discussed in hardcode-tray here: https://github.com/bilelmoussaoui/Hardcode-Tray/issues/577

**Comment by Dvlnst** [comment]
> The Papirus icon theme has a tray icon for this program. The tray icon for ElectronMail is hardcoded.
>
> The tray icon was discussed in hardcode-tray here: [bilelmoussaoui/Hardcode-Tray#577](https://github.com/bilelmoussaoui/Hardcode-Tray/issues/577)

Yes, I noticed this one yesterday. I tested Papirus icon theme and it's the same behaiviour, it needs a fixing script. About ElectronMail tray icon is hardcoded, I think that's how electron works. I'm not an expert, but if I'm not wrong, tray icon is created inside the BrowserWindow in running time and it forces you to specify a path for the icon. 
    

> Well, if it just uses `electron` as icon name we probably can't fix it without further ado. You'll probably need a fixing script, I don't know if there's a way to have it pick up a tray icon with `electronmail-tray` for instance.
> But then again I haven't got the slightest notion of how tray icon theming is done. Personally I try to get rid of tray icons as much as possible.

Excuse me for the late reply. I agree with you, I also try to avoid tray icons with Gnome (due to lack of official support), but in this machine I need Megasync app which expects tray support. The alternative is to keep the window opened and It's really annoying (a personal OCD haha).

If you can add at least a desktop icon for this app, I would be really grateful. I love numix icon theme (circle version) and I'm using it for a long time, so thanks for the work.

Kind regards.

**Comment by vladimiry** [comment]
> About ElectronMail tray icon is hardcoded, I think that's how electron works. I'm not an expert, but if I'm not wrong, tray icon is created inside the BrowserWindow in running time and it forces you to specify a path for the icon.

The icon is a static file packed into `asar` archive with the following path `/opt/ElectronMail/resources/app.asar/app/assets/icons/icon.png` (I guess base app path `/opt/ElectronMail/` might differ depending on the system). The app loads that icon and caches it into memory for further processing. Then when needed the app programmatically renders an unread counter on top of the base icon. The app also supports configuring the colors of the tray icon and unread counter (background and font colors separately). Such configuring is possible since the app generates the icon programmatically based on the static image file. Think like the app replaces colors on the fly preserving the original icon shaping.

On Arch Linux system the desktop file location is `/usr/share/applications/electron-mail.desktop` and here is its the contents:
```
[Desktop Entry]
Name=ElectronMail
Exec=/opt/ElectronMail/electron-mail %U
Terminal=false
Type=Application
Icon=electron-mail
StartupWMClass=ElectronMail
Comment=Unofficial ProtonMail Desktop App
Categories=Office;
```


By the way, icon sources located in https://github.com/vladimiry/ElectronMail/tree/master/src/assets/icon.
